x-pack/plugins/security_solution/public/detections/containers/detection_engine/lists/use_lists_config.tsx

```diff
@@ -29,7 +29,7 @@ export const useListsConfig = (): UseListsConfigReturn => {
   const hasIndexError = indexError != null;
   const needsIndexConfiguration =
     needsIndex && (canManageIndex === false || (canManageIndex === true && hasIndexError));
-  const needsConfiguration = !enabled || canWriteIndex === false || needsIndexConfiguration;
+  const needsConfiguration = !enabled || needsIndexConfiguration;
```

Comment:
Only suggestions I would have would be if we can have tests around this area that would be pretty nice to add to the documentation and help future maintainers.

```ts
return { canManageIndex, canWriteIndex, enabled, loading, needsConfiguration };
```

This looks really good in that later we can push this to the "needs permissions" and make actionable UI's to tell the user what they're missing exactly and how to fix it. So this was really nice to see here.

Reply:
I wholeheartedly agree on the tests. I had been waiting to write a functional test for "read-only lists user can use Detections," but in the interim I will add some unit tests around these hooks in a followup PR 👍